File: BankingApp/BankFrontend/FrontEnd/src/app/Interfaces/loan.ts
export interface Loan {
    id: number;
    amount: number;
    businessId: string;
    interestRate: number;
    dateLoaned: Date;
    loanPaid: Date;
    monthlyPay: number;
    amountPaid: number;
}

export interface LoanSchedule {
    date: string;
    payment: number;
    principal: number;
    interest: number;
    totalInterest: number;
    balance: number
}

File: BankingApp/BankBackend/DataAccess/Entities/Loan.cs
using System;
using System.Collections.Generic;

namespace DataAccess.Entities;

public partial class Loan
{
    public int Id { get; set; }

    public decimal? Amount { get; set; }

    public int? BusinessId { get; set; }

    public decimal? InterestRate { get; set; }

    public DateTime? DateLoaned { get; set; }

    public DateTime? LoanPaid { get; set; }

    public decimal? MonthlyPay { get; set; }

    public decimal? AmountPaid { get; set; }
}
